### PR TITLE
add -Wno-* compile option to suppress warnings

### PR DESCRIPTION
**Pull request by fmessmer**
I **also** would like to suppress the warnings from `libphidgets/ExternalProject` (see https://travis-ci.com/ipa320/cob_extern/jobs/253965974#L439-L517), but I can't figure out a way to do so

@lindemeier @benmaidel @ipa-mdl do you have any idea? any feedback on this pr?

### CODE REVIEW

File: libdlib/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ ExternalProject_Add(EP_${PROJECT_NAME}
     BINARY_DIR ${PROJECT_BINARY_DIR}/${PROJECT_NAME}-build
     
     CONFIGURE_COMMAND ${CMAKE_COMMAND} ${PROJECT_BINARY_DIR}/${PROJECT_NAME}-src/dlib
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . --config Release
+    BUILD_COMMAND ${CMAKE_COMMAND} --build . --config Release -- -Wno-deprecated-declarations -Wno-terminate
```
Comment (mathias-luedtke) [inline libdlib/CMakeLists.txt:20]: if you want to silence `libphidgets`, why are you specifying this for `libdlib`?

Comment (fmessmer) [reply libdlib/CMakeLists.txt:20]: I want to silence warnings from both packages...it works for libdlib...with this pr....
now I wanted to add a commit to this pr which silences the warnings issued by libphidgets...

in the end, I want them both silenced :wink:

Comment (mathias-luedtke) [reply libdlib/CMakeLists.txt:20]: Ah, "also" ;)

Comment (mathias-luedtke) [reply libdlib/CMakeLists.txt:20]: What happens, if you specify this for libphidgets as well?

Comment (fmessmer) [reply libdlib/CMakeLists.txt:20]: well, `libphidgets` uses [`make` as `BUILD_COMMAND`](https://github.com/ipa320/cob_extern/blob/indigo_dev/libphidgets/CMakeLists.txt#L17) while `libdlib` uses [`cmake` as `BUILD_COMMAND`](https://github.com/ipa320/cob_extern/blob/indigo_dev/libdlib/CMakeLists.txt#L20) - the same fix as for `cmake` does not work for `make`

Comment (mathias-luedtke) [reply libdlib/CMakeLists.txt:20]: I see. Normally, you would pass it in the configuration step anyway. E.g. by specifying `CXXFLAGS` and/or `CFLAGS`.

File: libphidgets/CMakeLists.txt [hunk 2]
```diff
@@ -11,7 +11,7 @@ ExternalProject_Add(EP_${PROJECT_NAME}
     URL https://www.phidgets.com/downloads/phidget21/libraries/linux/libphidget/libphidget_2.1.8.20151217.tar.gz
     URL_MD5 818ab2ff1de92ed9568a206e0e89657f
 
-    CONFIGURE_COMMAND "./configure"
+    CONFIGURE_COMMAND ${PROJECT_BINARY_DIR}/${PROJECT_NAME}-src/configure CFLAGS=-Wno-incompatible-pointer-types\ -Wno-deprecated-declarations
```
Comment (fmessmer) [inline libphidgets/CMakeLists.txt:14]: @ipa-mdl 
I found a way to pass the desired `-Wno-*` arguments to the `libphigets/configure` script

**Comment by ipa-rmb** [comment]
+1

**Pull request merged**